internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/server/VeniceServerTest.java

```diff
@@ -182,6 +183,33 @@ public void testCheckBeforeJointClusterBeforeHelixInitializingCluster() throws E
     }
   }
 
+  @Test
+  public void testShutdown() {
+    try (VeniceClusterWrapper cluster = ServiceFactory.getVeniceCluster(1, 1, 0)) {
+      VeniceServerWrapper server = cluster.getVeniceServers().get(0);
+      StorageService storageService = server.getVeniceServer().getStorageService();
+      StorageEngineRepository repository = storageService.getStorageEngineRepository();
+      Assert
+          .assertTrue(repository.getAllLocalStorageEngines().isEmpty(), "New node should not have any storage engine.");
+
+      // Create a storage engine.
+      String storeName = Version.composeKafkaTopic(cluster.createStore(1), 1);
+      storageService.openStoreForNewPartition(
+          server.getVeniceServer().getConfigLoader().getStoreConfig(storeName),
+          1,
+          () -> null);
+      Assert.assertEquals(
+          repository.getAllLocalStorageEngines().size(),
+          1,
+          "We have created one storage engine for store: " + storeName);
```

Comment:
You shouldn't have to do this. `cluster.createStore` should be doing this under the hood eventually.